LendingClub is a US peer-to-peer lending company, headquartered in San Francisco, California. It was the first peer-to-peer lender to register its offerings as securities with the Securities and Exchange Commission (SEC), and to offer loan trading on a secondary market. LendingClub is the world's largest peer-to-peer lending platform.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import pandas as pd
# prueba github
# Cargar los archivos en DataFrames
lc = pd.read_csv('../data/raw/lending_club_loan_two.csv')
lc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 396030 entries, 0 to 396029
Data columns (total 27 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   loan_amnt             396030 non-null  float64
 1   term                  396030 non-null  object 
 2   int_rate              396030 non-null  float64
 3   installment           396030 non-null  float64
 4   grade                 396030 non-null  object 
 5   sub_grade             396030 non-null  object 
 6   emp_title             373103 non-null  object 
 7   emp_length            377729 non-null  object 
 8   home_ownership        396030 non-null  object 
 9   annual_inc            396030 non-null  float64
 10  verification_status   396030 non-null  object 
 11  issue_d               396030 non-null  object 
 12  loan_status           396030 non-null  object 
 13  purpose               396030 non-null  object 
 14  title                 394274 non-null  object 
 15  

In [3]:
pd.set_option('display.max_columns', None)
lc.head()

loan_amnt        term  int_rate  installment grade sub_grade  \
0    10000.0   36 months     11.44       329.48     B        B4   
1     8000.0   36 months     11.99       265.68     B        B5   
2    15600.0   36 months     10.49       506.97     B        B3   
3     7200.0   36 months      6.49       220.65     A        A2   
4    24375.0   60 months     17.27       609.33     C        C5   

                 emp_title emp_length home_ownership  annual_inc  \
0                Marketing  10+ years           RENT    117000.0   
1          Credit analyst     4 years       MORTGAGE     65000.0   
2             Statistician   < 1 year           RENT     43057.0   
3          Client Advocate    6 years           RENT     54000.0   
4  Destiny Management Inc.    9 years       MORTGAGE     55000.0   

  verification_status   issue_d  loan_status             purpose  \
0        Not Verified  Jan-2015   Fully Paid            vacation   
1        Not Verified  Jan-2015   Fully Paid  debt_consolidation   
2     Source Verified  Jan-2015   Fully Paid         credit_card   
3        Not Verified  Nov-2014   Fully Paid         credit_card   
4            Verified  Apr-2013  Charged Off         credit_card   

                     title    dti earliest_cr_line  open_acc  pub_rec  \
0                 Vacation  26.24         Jun-1990      16.0      0.0   
1       Debt consolidation  22.05         Jul-2004      17.0      0.0   
2  Credit card refinancing  12.79         Aug-2007      13.0      0.0   
3  Credit card refinancing   2.60         Sep-2006       6.0      0.0   
4    Credit Card Refinance  33.95         Mar-1999      13.0      0.0   

   revol_bal  revol_util  total_acc initial_list_status application_type  \
0    36369.0        41.8       25.0                   w       INDIVIDUAL   
1    20131.0        53.3       27.0                   f       INDIVIDUAL   
2    11987.0        92.2       26.0                   f       INDIVIDUAL   
3     5472.0        21.5       13.0                   f       INDIVIDUAL   
4    24584.0        69.8       43.0                   f       INDIVIDUAL   

   mort_acc  pub_rec_bankruptcies  \
0       0.0                   0.0   
1       3.0                   0.0   
2       0.0                   0.0   
3       0.0                   0.0   
4       1.0                   0.0   

                                             address  
0     0174 Michelle Gateway\r\nMendozaberg, OK 22690  
1  1076 Carney Fort Apt. 347\r\nLoganmouth, SD 05113  
2  87025 Mark Dale Apt. 269\r\nNew Sabrina, WV 05113  
3            823 Reid Ford\r\nDelacruzside, MA 00813  
4             679 Luna Roads\r\nGreggshire, VA 11650

loan_amnt: Monto del préstamo solicitado.
term: Plazo del préstamo en meses.
int_rate: Tasa de interés del préstamo.
installment: Pago mensual del préstamo.
grade: Grado asignado al préstamo por LendingClub (A, B, C, etc.).
sub_grade: Subgrado asignado al préstamo por LendingClub (A1, A2, B1, etc.).
emp_title: Título laboral del prestatario.
emp_length: Antigüedad laboral del prestatario en años.
home_ownership: Estado de propiedad de la vivienda del prestatario (RENT, OWN, MORTGAGE, OTHER).
annual_inc: Ingreso anual del prestatario.
verification_status: Estado de verificación del ingreso del prestatario.
issue_d: Fecha en la que se emitió el préstamo.
loan_status: Estado actual del préstamo (Fully Paid, Charged Off, etc.).
purpose: Propósito del préstamo.
title: Título del préstamo proporcionado por el prestatario.
dti: Relación entre las deudas y el ingreso del prestatario.
earliest_cr_line: Fecha en que se abrió la primera línea de crédito del prestatario.
open_acc: Número de líneas de crédito abiertas en el archivo del prestatario.
pub_rec: Número de registros públicos desfavorables.
revol_bal: Saldo total de las cuentas de crédito renovable.
revol_util: Tasa de utilización de las cuentas de crédito renovable.
total_acc: Número total de cuentas de crédito del prestatario.
initial_list_status: Estado inicial de la lista del préstamo (W, F).
application_type: Tipo de aplicación (INDIVIDUAL, JOINT).
mort_acc: Número de cuentas hipotecarias.
pub_rec_bankruptcies: Número de quiebras en los registros públicos.
address: Dirección del prestatario.

In [4]:
lc.describe()

loan_amnt       int_rate    installment    annual_inc  \
count  396030.000000  396030.000000  396030.000000  3.960300e+05   
mean    14113.888089      13.639400     431.849698  7.420318e+04   
std      8357.441341       4.472157     250.727790  6.163762e+04   
min       500.000000       5.320000      16.080000  0.000000e+00   
25%      8000.000000      10.490000     250.330000  4.500000e+04   
50%     12000.000000      13.330000     375.430000  6.400000e+04   
75%     20000.000000      16.490000     567.300000  9.000000e+04   
max     40000.000000      30.990000    1533.810000  8.706582e+06   

                 dti       open_acc        pub_rec     revol_bal  \
count  396030.000000  396030.000000  396030.000000  3.960300e+05   
mean       17.379514      11.311153       0.178191  1.584454e+04   
std        18.019092       5.137649       0.530671  2.059184e+04   
min         0.000000       0.000000       0.000000  0.000000e+00   
25%        11.280000       8.000000       0.000000  6.025000e+03   
50%        16.910000      10.000000       0.000000  1.118100e+04   
75%        22.980000      14.000000       0.000000  1.962000e+04   
max      9999.000000      90.000000      86.000000  1.743266e+06   

          revol_util      total_acc       mort_acc  pub_rec_bankruptcies  
count  395754.000000  396030.000000  358235.000000         395495.000000  
mean       53.791749      25.414744       1.813991              0.121648  
std        24.452193      11.886991       2.147930              0.356174  
min         0.000000       2.000000       0.000000              0.000000  
25%        35.800000      17.000000       0.000000              0.000000  
50%        54.800000      24.000000       1.000000              0.000000  
75%        72.900000      32.000000       3.000000              0.000000  
max       892.300000     151.000000      34.000000              8.000000

In [5]:
lc['issue_d'].dtype

dtype('O')

In [6]:
lc['issue_d'].max()

'Sep-2016'

In [7]:
lc['issue_d'].min()

'Apr-2008'

In [8]:
lc['loan_status'].unique()

array(['Fully Paid', 'Charged Off'], dtype=object)

In [9]:
lc['loan_status'].value_counts(normalize=True)

loan_status
Fully Paid     0.803871
Charged Off    0.196129
Name: proportion, dtype: float64

In [10]:
lc['issue_d'].dtype

dtype('O')

In [11]:
lc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 396030 entries, 0 to 396029
Data columns (total 27 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   loan_amnt             396030 non-null  float64
 1   term                  396030 non-null  object 
 2   int_rate              396030 non-null  float64
 3   installment           396030 non-null  float64
 4   grade                 396030 non-null  object 
 5   sub_grade             396030 non-null  object 
 6   emp_title             373103 non-null  object 
 7   emp_length            377729 non-null  object 
 8   home_ownership        396030 non-null  object 
 9   annual_inc            396030 non-null  float64
 10  verification_status   396030 non-null  object 
 11  issue_d               396030 non-null  object 
 12  loan_status           396030 non-null  object 
 13  purpose               396030 non-null  object 
 14  title                 394274 non-null  object 
 15  

In [12]:
# OJO PARA QUEDARME CON UN AÑO SOLO........ACHICAR MUESTRA Y SEA MAS MANEJABLE


lc['issue_d'] = pd.to_datetime(lc['issue_d'], format='%b-%Y')

# Filtrar el DataFrame para las fechas específicas
lc_2012 = lc[(lc['issue_d'] >= '2012-01-01') & (lc['issue_d'] <= '2012-12-31')]

In [13]:
lc_2012['loan_status'].value_counts(normalize=True)

loan_status
Fully Paid     0.835202
Charged Off    0.164798
Name: proportion, dtype: float64

In [14]:

# OJO CON ESTO..........................

w_p = lc_2012.loan_status.value_counts()[0] / lc_2012.shape[0]
w_n = lc_2012.loan_status.value_counts()[1] / lc_2012.shape[0]

print(f"Weight of positive values {w_p}")
print(f"Weight of negative values {w_n}")

Weight of positive values 0.835202174651716
Weight of negative values 0.16479782534828408


In [15]:
lc_2012.head()

loan_amnt        term  int_rate  installment grade sub_grade  \
7     13000.0   36 months     11.14       426.47     B        B2   
9     26300.0   36 months     16.29       928.40     C        C5   
14    25975.0   36 months      6.62       797.53     A        A2   
75    13200.0   36 months     10.74       430.53     B        B2   
80    19000.0   36 months     12.12       632.17     B        B3   

                        emp_title emp_length home_ownership  annual_inc  \
7                    Office Depot  10+ years           RENT     46000.0   
9              Regado Biosciences    3 years       MORTGAGE    115000.0   
14      Pullman Regional Hospital    9 years       MORTGAGE     65000.0   
75  Brookdale University Hospital     1 year           RENT     45000.0   
80         Broward Sheriff Office    8 years           RENT     48000.0   

   verification_status    issue_d loan_status             purpose  \
7         Not Verified 2012-09-01  Fully Paid         credit_card   
9             Verified 2012-04-01  Fully Paid  debt_consolidation   
14            Verified 2012-10-01  Fully Paid      small_business   
75     Source Verified 2012-03-01  Fully Paid  debt_consolidation   
80     Source Verified 2012-07-01  Fully Paid  debt_consolidation   

                   title    dti earliest_cr_line  open_acc  pub_rec  \
7   No More Credit Cards  26.87         Sep-1994      11.0      0.0   
9     Debt Consolidation  23.69         Dec-1997      13.0      0.0   
14              Business   9.18         May-2001       9.0      0.0   
75  Honost Consolidation  12.80         May-1994       8.0      0.0   
80             Debt Loan  29.76         Jun-1994      12.0      0.0   

    revol_bal  revol_util  total_acc initial_list_status application_type  \
7     13425.0        64.5       15.0                   f       INDIVIDUAL   
9     22171.0        82.4       37.0                   f       INDIVIDUAL   
14     5140.0         7.2       35.0                   f       INDIVIDUAL   
75    10665.0        75.6       10.0                   f       INDIVIDUAL   
80    30071.0        46.5       28.0                   f       INDIVIDUAL   

    mort_acc  pub_rec_bankruptcies  \
7        0.0                   0.0   
9        1.0                   0.0   
14       6.0                   0.0   
75       0.0                   0.0   
80       0.0                   0.0   

                                              address  
7                         USCGC Nunez\r\nFPO AE 30723  
9              3390 Luis Rue\r\nMauricestad, VA 00813  
14  0472 Weber Manors Apt. 660\r\nLake Andrew, NH ...  
75  1741 Jackson Street Suite 615\r\nNew Jenny, MA...  
80  58673 Foster Freeway\r\nPort Meredithville, NY...

In [16]:
lc_2012.info()

<class 'pandas.core.frame.DataFrame'>
Index: 41202 entries, 7 to 396028
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   loan_amnt             41202 non-null  float64       
 1   term                  41202 non-null  object        
 2   int_rate              41202 non-null  float64       
 3   installment           41202 non-null  float64       
 4   grade                 41202 non-null  object        
 5   sub_grade             41202 non-null  object        
 6   emp_title             38725 non-null  object        
 7   emp_length            39691 non-null  object        
 8   home_ownership        41202 non-null  object        
 9   annual_inc            41202 non-null  float64       
 10  verification_status   41202 non-null  object        
 11  issue_d               41202 non-null  datetime64[ns]
 12  loan_status           41202 non-null  object        
 13  purpose             

5   sub_grade             41202 non-null  object        
 6   emp_title             38725 non-null  object ...............OJO       
 7   emp_length            39691 non-null  object .................OJO       
 8   home_ownership        41202 non-null  object        
 9   annual_inc            41202 non-null  float64       
 10  verification_status   41202 non-null  object        
 11  issue_d               41202 non-null  datetime64[ns]
 
 12  loan_status           41202 non-null  object        
 13  purpose               41202 non-null  object        
 14  title                 41201 non-null  object  ...................OJO      
 15  dti                   41202 non-null  float64       
 16  earliest_cr_line      41202 non-null  object        
 17  open_acc              41202 non-null  float64       
 18  pub_rec               41202 non-null  float64       
 19  revol_bal             41202 non-null  float64       
 20  revol_util            41169 non-null  float64 ...................OJO      
 21  total_acc             41202 non-null  float64       
 22  initial_list_status   41202 non-null  object        
 23  application_type      41202 non-null  object        
 24  mort_acc              35361 non-null  float64 ..................OJO      
 25  pub_rec_bankruptcies  41202 non-null  float64       
 26  address               41202 non-null  object    

In [17]:
lc_2012.describe()

loan_amnt      int_rate   installment    annual_inc  \
count  41202.000000  41202.000000  41202.000000  4.120200e+04   
mean   13133.800544     13.459286    416.806544  6.921865e+04   
min     1000.000000      6.000000     21.620000  4.800000e+03   
25%     7000.000000     10.160000    232.382500  4.200000e+04   
50%    11500.000000     13.110000    369.320000  6.000000e+04   
75%    18000.000000     16.290000    544.170000  8.366150e+04   
max    35000.000000     24.890000   1388.450000  7.141778e+06   
std     7949.106060      4.303924    246.072616  6.172477e+04   

                             issue_d           dti      open_acc  \
count                          41202  41202.000000  41202.000000   
mean   2012-07-19 01:23:15.019659520     16.600915     10.599049   
min              2012-01-01 00:00:00      0.000000      1.000000   
25%              2012-05-01 00:00:00     10.920000      7.000000   
50%              2012-08-01 00:00:00     16.410000     10.000000   
75%              2012-10-01 00:00:00     22.010000     13.000000   
max              2012-12-01 00:00:00     34.990000     49.000000   
std                              NaN      7.589258      4.454512   

            pub_rec      revol_bal    revol_util     total_acc      mort_acc  \
count  41202.000000   41202.000000  41169.000000  41202.000000  35361.000000   
mean       0.029489   14870.548323     57.677532     23.492937      1.590594   
min        0.000000       0.000000      0.000000      3.000000      0.000000   
25%        0.000000    6603.000000     41.000000     15.000000      0.000000   
50%        0.000000   11676.000000     60.700000     22.000000      1.000000   
75%        0.000000   19257.000000     77.100000     30.000000      3.000000   
max        5.000000  975800.000000    104.400000     99.000000     24.000000   
std        0.177847   14617.625472     24.265878     10.919299      2.165704   

       pub_rec_bankruptcies  
count          41202.000000  
mean               0.023907  
min                0.000000  
25%                0.000000  
50%                0.000000  
75%                0.000000  
max                5.000000  
std                0.156527

In [18]:
# Crear un DataFrame vacío para almacenar los resultados
resultados_por_año = pd.DataFrame()

# Iterar sobre los años
for año in range(2009, 2016):
    # Filtrar el DataFrame para el año específico
    lc_año = lc[(lc['issue_d'].dt.year == año)]
    
    # Guardar el DataFrame filtrado en el DataFrame de resultados
    resultados_por_año = pd.concat([resultados_por_año, lc_año])
    
    # Mostrar la distribución de 'loan_status' para el año específico
    print(f'Loan Status Distribution for {año}:')
    print(lc_año['loan_status'].value_counts(normalize=True))
    print('\n')

# Mostrar el DataFrame completo de resultados
print('DataFrame completo con datos por año:')
print(resultados_por_año)

Loan Status Distribution for 2009:
loan_status
Fully Paid     0.877418
Charged Off    0.122582
Name: proportion, dtype: float64


Loan Status Distribution for 2010:
loan_status
Fully Paid     0.867898
Charged Off    0.132102
Name: proportion, dtype: float64


Loan Status Distribution for 2011:
loan_status
Fully Paid     0.847548
Charged Off    0.152452
Name: proportion, dtype: float64


Loan Status Distribution for 2012:
loan_status
Fully Paid     0.835202
Charged Off    0.164798
Name: proportion, dtype: float64


Loan Status Distribution for 2013:
loan_status
Fully Paid     0.8426
Charged Off    0.1574
Name: proportion, dtype: float64


Loan Status Distribution for 2014:
loan_status
Fully Paid     0.76889
Charged Off    0.23111
Name: proportion, dtype: float64


Loan Status Distribution for 2015:
loan_status
Fully Paid     0.750987
Charged Off    0.249013
Name: proportion, dtype: float64


DataFrame completo con datos por año:
        loan_amnt        term  int_rate  installment grade

In [19]:
# Crear un diccionario para almacenar el conteo por año
conteo_por_año = {}

# Iterar sobre los años
for año in range(2009, 2016):
    # Filtrar el DataFrame para el año específico
    lc_año = lc[(lc['issue_d'].dt.year == año)]
    
    # Almacenar el conteo en el diccionario
    conteo_por_año[año] = len(lc_año)

# Mostrar el conteo por año
for año, conteo in conteo_por_año.items():
    print(f'Número de elementos para el año {año}: {conteo}')

Número de elementos para el año 2009: 3826
Número de elementos para el año 2010: 9258
Número de elementos para el año 2011: 17435
Número de elementos para el año 2012: 41202
Número de elementos para el año 2013: 97662
Número de elementos para el año 2014: 102860
Número de elementos para el año 2015: 94264


In [20]:

lc['issue_d'] = pd.to_datetime(lc['issue_d'], format='%b-%Y')

# Filtrar el DataFrame para las fechas específicas
lc_2015 = lc[(lc['issue_d'] >= '2015-01-01') & (lc['issue_d'] <= '2015-12-31')]

In [21]:
lc_2015


loan_amnt        term  int_rate  installment grade sub_grade  \
0         10000.0   36 months     11.44       329.48     B        B4   
1          8000.0   36 months     11.99       265.68     B        B5   
2         15600.0   36 months     10.49       506.97     B        B3   
5         20000.0   36 months     13.33       677.07     C        C3   
6         18000.0   36 months      5.32       542.07     A        A1   
...           ...         ...       ...          ...   ...       ...   
396017     9600.0   36 months      9.99       309.72     B        B3   
396020    10000.0   36 months      9.76       321.55     B        B3   
396022    12000.0   36 months     12.29       400.24     C        C1   
396025    10000.0   60 months     10.99       217.38     B        B4   
396026    21000.0   36 months     12.29       700.42     C        C1   

                            emp_title emp_length home_ownership  annual_inc  \
0                           Marketing  10+ years           RENT    117000.0   
1                     Credit analyst     4 years       MORTGAGE     65000.0   
2                        Statistician   < 1 year           RENT     43057.0   
5                       HR Specialist  10+ years       MORTGAGE     86788.0   
6       Software Development Engineer    2 years       MORTGAGE    125000.0   
...                               ...        ...            ...         ...   
396017                        teacher    8 years       MORTGAGE     41000.0   
396020           Retirement Counselor  10+ years           RENT     40000.0   
396022      Data Center Specialist II     1 year           RENT     52100.0   
396025               licensed bankere    2 years           RENT     40000.0   
396026                          Agent    5 years       MORTGAGE    110000.0   

       verification_status    issue_d loan_status             purpose  \
0             Not Verified 2015-01-01  Fully Paid            vacation   
1             Not Verified 2015-01-01  Fully Paid  debt_consolidation   
2          Source Verified 2015-01-01  Fully Paid         credit_card   
5                 Verified 2015-09-01  Fully Paid  debt_consolidation   
6          Source Verified 2015-09-01  Fully Paid    home_improvement   
...                    ...        ...         ...                 ...   
396017     Source Verified 2015-05-01  Fully Paid         credit_card   
396020        Not Verified 2015-12-01  Fully Paid  debt_consolidation   
396022     Source Verified 2015-10-01  Fully Paid  debt_consolidation   
396025     Source Verified 2015-10-01  Fully Paid  debt_consolidation   
396026     Source Verified 2015-02-01  Fully Paid  debt_consolidation   

                          title    dti earliest_cr_line  open_acc  pub_rec  \
0                      Vacation  26.24         Jun-1990      16.0      0.0   
1            Debt consolidation  22.05         Jul-2004      17.0      0.0   
2       Credit card refinancing  12.79         Aug-2007      13.0      0.0   
5            Debt consolidation  16.31         Jan-2005       8.0      0.0   
6              Home improvement   1.36         Aug-2005       8.0      0.0   
...                         ...    ...              ...       ...      ...   
396017  Credit card refinancing  23.65         Sep-2000       9.0      0.0   
396020       Debt consolidation  23.40         Jan-1988       9.0      0.0   
396022       Debt consolidation  17.28         Oct-2004       6.0      0.0   
396025       Debt consolidation  15.63         Nov-2004       6.0      0.0   
396026       Debt consolidation  21.45         Feb-2006       6.0      0.0   

        revol_bal  revol_util  total_acc initial_list_status application_type  \
0         36369.0        41.8       25.0                   w       INDIVIDUAL   
1         20131.0        53.3       27.0                   f       INDIVIDUAL   
2         11987.0        92.2       26.0                   f       INDIVIDUAL   
5         25757.0       100.6       23.0                

In [22]:
lc['issue_d'] = pd.to_datetime(lc['issue_d'], format='%b-%Y')

# Filtrar el DataFrame para las fechas específicas
lc_2011 = lc[(lc['issue_d'] >= '2011-01-01') & (lc['issue_d'] <= '2011-12-31')]

In [23]:
lc_2011.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17435 entries, 22 to 396021
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   loan_amnt             17435 non-null  float64       
 1   term                  17435 non-null  object        
 2   int_rate              17435 non-null  float64       
 3   installment           17435 non-null  float64       
 4   grade                 17435 non-null  object        
 5   sub_grade             17435 non-null  object        
 6   emp_title             16285 non-null  object        
 7   emp_length            16831 non-null  object        
 8   home_ownership        17435 non-null  object        
 9   annual_inc            17435 non-null  float64       
 10  verification_status   17435 non-null  object        
 11  issue_d               17435 non-null  datetime64[ns]
 12  loan_status           17435 non-null  object        
 13  purpose            

In [24]:
lc['issue_d'] = pd.to_datetime(lc['issue_d'], format='%b-%Y')

# Filtrar el DataFrame para las fechas específicas
lc_2010 = lc[(lc['issue_d'] >= '2010-01-01') & (lc['issue_d'] <= '2010-12-31')]

In [25]:
lc_2010.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9258 entries, 25 to 396029
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   loan_amnt             9258 non-null   float64       
 1   term                  9258 non-null   object        
 2   int_rate              9258 non-null   float64       
 3   installment           9258 non-null   float64       
 4   grade                 9258 non-null   object        
 5   sub_grade             9258 non-null   object        
 6   emp_title             8713 non-null   object        
 7   emp_length            9003 non-null   object        
 8   home_ownership        9258 non-null   object        
 9   annual_inc            9258 non-null   float64       
 10  verification_status   9258 non-null   object        
 11  issue_d               9258 non-null   datetime64[ns]
 12  loan_status           9258 non-null   object        
 13  purpose             

In [ ]:
plt.figure(figsize=(12,6))
sns.heatmap(df.corr(), annot=True, cmap="coolwarm", vmin=-1)

In [26]:
# OJO PARA BALANCEAR LA MUESTRA

# from sklearn.utils import resample


# data_majority = df[df['SeriousDlqin2yrs'] == 0]
# data_minority = df[df['SeriousDlqin2yrs'] == 1]

# data_majority_downsampled = resample(data_majority, replace=False, n_samples=len(data_minority), random_state=42)

# data_balanced = pd.concat([data_majority_downsampled, data_minority])